### PR TITLE
OCPBUGS-43882: improve eBPF API reference docs

### PR DESCRIPTION
**Pull request by skrthomas**
Version(s):

4.12+
Issue:

https://issues.redhat.com/browse/OCPBUGS-43882
Link to docs preview:

QE review:
- [X] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by skrthomas** [comment]
@jotak @memodi PTAL at these updates to the FlowCollector API reference. Thank you :)

**Review by jotak — APPROVED** [review]
LGTM thanks @skrthomas !

**Review by memodi — APPROVED** [review]
/label qe-approved

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 1]
```diff
@@ -386,57 +386,58 @@ Examples: `10.10.10.0/24` or `100:100:100:100::/64`
...
-| `ports` defines the ports to filter flows by. It is used both for source and destination ports.
+| `ports` optionally defines the ports to filter flows by. It is used both for source and destination ports.
 To filter a single port, set a single port as an integer value. For example, `ports: 80`.
 To filter a range of ports, use a "start-end" range in string format. For example, `ports: "80-100"`.
 To filter two ports, use a "port1,port2" in string format. For example, `ports: "80,100"`.
 
 | `protocol`
 | `string`
-| `protocol` defines the protocol to filter flows by.
+| `protocol` optionally defines a protocol to filter flows by. The available options are `TCP`, `UDP`, `ICMP`, `ICMPv6` and `SCTP`.
```
Comment (jeana-redhat) [inline modules/network-observability-flowcollector-api-specifications.adoc:428]:
```suggestion
| `protocol` optionally defines a protocol to filter flows by. The available options are `TCP`, `UDP`, `ICMP`, `ICMPv6`, and `SCTP`.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 2]
```diff
@@ -386,57 +386,58 @@ Examples: `10.10.10.0/24` or `100:100:100:100::/64`
...
+| `sourcePorts` optionally defines the source ports to filter flows by.
 To filter a single port, set a single port as an integer value. For example, `sourcePorts: 80`.
 To filter a range of ports, use a "start-end" range in string format. For example, `sourcePorts: "80-100"`.
 To filter two ports, use a "port1,port2" in string format. For example, `ports: "80,100"`.
 
 | `tcpFlags`
 | `string`
-| `tcpFlags` defines the TCP flags to filter flows by.
+| `tcpFlags` optionally defines TCP flags to filter flows by.
+In addition to the standard flags (RFC-9293), you can also filter by one of the three following combinations: `SYN-ACK`, `FIN-ACK` and `RST-ACK`.
```
Comment (jeana-redhat) [inline modules/network-observability-flowcollector-api-specifications.adoc:440]:
```suggestion
In addition to the standard flags (RFC-9293), you can also filter by one of the three following combinations: `SYN-ACK`, `FIN-ACK`, and `RST-ACK`.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 3]
```diff
@@ -461,7 +462,7 @@ Type::
 | `disableAlerts` is a list of alerts that should be disabled.
 Possible values are: +
 
-`NetObservDroppedFlows` is triggered when the eBPF agent is missing packets or flows, such as when the eBPF hashmap is busy or full, or the capacity limiter is triggered. +
+`NetObservDroppedFlows`, which is triggered when the eBPF agent is missing packets or flows, such as when the BPF hashmap is busy or full, or the capacity limiter is being triggered. +
```
Comment (jeana-redhat) [inline modules/network-observability-flowcollector-api-specifications.adoc:465]: Possible to split the example cases off as a second sentence?

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 4]
```diff
@@ -978,7 +979,7 @@ Required::
 
 | `type`
 | `string`
-| `type` selects the type of exporters. The available options are `Kafka` and `IPFIX`.
+| `type` selects the type of exporters. The available options are `Kafka`, `IPFIX` and `OpenTelemetry`.
```
Comment (jeana-redhat) [inline modules/network-observability-flowcollector-api-specifications.adoc:982]:
```suggestion
| `type` selects the type of exporters. The available options are `Kafka`, `IPFIX`, and `OpenTelemetry`.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 5]
```diff
@@ -2538,7 +2539,7 @@ configuration, you can disable it and install your own instead.
 | `boolean`
 | Set `enable` to `true` to deploy network policies on the namespaces used by Network Observability (main and privileged). It is disabled by default.
 These network policies better isolate the Network Observability components to prevent undesired connections to them.
-Either enable it, or create your own network policy for Network Observability.
+We recommend you either enable it, or create your own network policy for Network Observability.
```
Comment (jeana-redhat) [inline modules/network-observability-flowcollector-api-specifications.adoc:2542]: I've seen caution about using "recommend" language in docs. Can this be stated in a way that avoids it?

**Review by jeana-redhat — COMMENTED** [review]
a few little nits and questions, overall LGTM

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

File: modules/network-observability-flowcollector-api-specifications.adoc (hunk 3 above)
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:465]: Thanks but I think its OK as it is.

File: modules/network-observability-flowcollector-api-specifications.adoc (hunk 5 above)
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:2542]: I think we can maybe write instead: To increase the security of connections, enable this option or create your own network policy.

I've added this in a comment in @jotak 's PR https://github.com/netobserv/network-observability-operator/pull/827#pullrequestreview-2428121065, along with your other feedbacks, so we can make sure the upstream/downstream are synchronized. Thanks for your review Jeana!

**Pull request merged**